### PR TITLE
added several tests

### PR DESCRIPTION
**Pull request by maxSedelnikov**

### CODE REVIEW

File: src/components/Alert/Alert.test.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import renderWithRedux from '../../testUtils/renderWithRedux';
+import Alert from './Alert';
+
+describe('Alert', () => {
+  it('shows alert', () => {
+    const { getByTestId } = renderWithRedux(<Alert />, {
+      initialState: {
+        alert: {
+          showAlert: true,
+          type: 'error',
```
Comment (albenik) [inline src/components/Alert/Alert.test.js:11]: Where is the field used?
I've checked Alert component itself and didn't find any place.

File: src/components/Alert/Alert.test.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      },
+    });
+    const alert = getByTestId('alert');
+
+    expect(alert).toHaveClass('active');
+    expect(alert).toHaveTextContent(/Some error appeared/i);
+  });
+
+  it('hides alert', () => {
+    const { getByTestId } = renderWithRedux(<Alert />);
```
Comment (albenik) [inline src/components/Alert/Alert.test.js:23]: In this test you actually doesn't hides a visible alert, but just renders it in initial (hidden) state.

File: src/components/Forms/AddToCartForm/AddToCartForm.test.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    const dispatch = jest.fn();
+    useDispatch.mockReturnValue(dispatch);
+    const { getByRole, getByText, queryByText } = render(<AddToCartForm />);
+    const productImageInput = getByRole('textbox', {
+      name: labelImg,
+    });
+
+    userEvent.type(productImageInput, 'https://image');
+    expect(getByText(invalidImgMsg)).toBeInTheDocument();
+    userEvent.type(productImageInput, 'https://image.png');
```
Comment (albenik) [inline src/components/Forms/AddToCartForm/AddToCartForm.test.js:58]: Probably this URL is incorrect too :)
Did you use `image.png` as a domain name or as a file name?

File: src/pages/Cart/Cart.test.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    expect(loader).toBeNull();
+    expect(heading).toBeInTheDocument();
+    expect(numberOfItemsInCart).toHaveTextContent('0');
+    expect(cartSum).toHaveTextContent('0.00 $');
+    expect(finishOrderBtn).toBeDisabled();
+  });
+
+  it('renders cart with items', () => {
+    const { getByTestId, getAllByTestId } = renderWithRedux(<Cart />, {
+      initialState: {
```
Comment (albenik) [inline src/pages/Cart/Cart.test.js:59]: Is this state correct? I see the items but `isEmptyCartSet: true`.

File: src/pages/Home/Home.test.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import Home from './Home';
+
+describe('Home', () => {
+  it('renders home page', () => {
```
Comment (albenik) [inline src/pages/Home/Home.test.js:6]: (imho) In the read world this is a useless test.
It's too atomic. Requires resources to write and mantain it but don't test anything valuable.
Consider use the snapshot tests.

File: src/components/Alert/Alert.test.js (hunk 1 above)
Comment (maxSedelnikov) [reply src/components/Alert/Alert.test.js:13]: the component alert is rendered in the app component https://prnt.sc/115o2km
and it is being triggered e.g when we could not fetch cart items https://prnt.sc/115o33w
updated the test in the next commit https://github.com/maxSedelnikov/react-shopping-app/pull/6/commits/c759e7206dd9006634852446e70ed64047e50a1d

File: src/components/Alert/Alert.test.js (hunk 2 above)
Comment (maxSedelnikov) [reply src/components/Alert/Alert.test.js]: solved here https://github.com/maxSedelnikov/react-shopping-app/pull/6/commits/c759e7206dd9006634852446e70ed64047e50a1d

File: src/components/Forms/AddToCartForm/AddToCartForm.test.js (hunk 3 above)
Comment (maxSedelnikov) [reply src/components/Forms/AddToCartForm/AddToCartForm.test.js]: I used the url as a domain name, solved https://github.com/maxSedelnikov/react-shopping-app/pull/6/commits/c759e7206dd9006634852446e70ed64047e50a1d

File: src/pages/Cart/Cart.test.js (hunk 4 above)
Comment (maxSedelnikov) [reply src/pages/Cart/Cart.test.js:114]: you right, solved and reworked here https://github.com/maxSedelnikov/react-shopping-app/pull/6/commits/c759e7206dd9006634852446e70ed64047e50a1d

File: src/pages/Home/Home.test.js (hunk 5 above)
Comment (maxSedelnikov) [reply src/pages/Home/Home.test.js]: removed not needed tests for now

File: src/components/CartItems/CartItems.test.js [hunk 6]
```diff
@@ -12,7 +11,7 @@ describe('CartItems', () => {
       {
         id: '-MXGc7d2QzP8TTDWgBX_',
         name: 'test',
-        pictureUrl: 'https://test.img',
+        pictureUrl: 'https://ipsumimage.appspot.com/140x100.png',
```
Comment (albenik) [inline src/components/CartItems/CartItems.test.js:15]: Okay, but did you correct regexp?
Or 'https://test.img' is still valid?

Comment (maxSedelnikov) [reply src/components/CartItems/CartItems.test.js:15]: yes i did)

**Pull request merged**